src/lib/y2caasp/clients/admin_role_dialog.rb

```diff
@@ -45,8 +55,47 @@ def contents
       @content = HSquash(
         MinWidth(50,
           # preselect the servers from the DHCP response
-          Y2Caasp::Widgets::NtpServer.new(dhcp_ntp_servers))
+          Y2Caasp::Widgets::NtpServer.new(ntp_servers))
       )
     end
+
+  private
+
+    #
+    # Propose the NTP servers from the DHCP response, fallback to a random
+    # machine from the ntp.org pool if enabled in control.xml.
+    #
+    # @return [Array<String>] proposed NTP servers, empty if nothing suitable found
+    #
+    def ntp_servers
+      # TODO: use Yast::NtpClient.ntp_conf if configured
+      # to better handle going back
+      servers = dhcp_ntp_servers
+      servers = ntp_fallback if servers.empty?
+
+      servers
+    end
+
+    #
+    # The fallback servers for NTP configuration
+    #
+    # @return [Array<String>] the fallback servers, empty if disabled in control.xml
+    #
+    def ntp_fallback
+      # propose the fallback when enabled in control file
+      return [] unless Yast::ProductFeatures.GetBooleanFeature("globals", "default_ntp_setup")
+
+      # copied from timezone/dialogs.rb:
```

Comment:
NP: Just a reminder of comment https://github.com/yast/yast-country/pull/186/#discussion_r232630094

We should have this in its own _importable(?)_ method.

Reply:
Yes, this should be later shared somehow...